Sources/SwiftDocC/Model/ParametersAndReturnValidator.swift

```diff
@@ -0,0 +1,473 @@
+        }
+        
+        // Diagnose if the symbol had documented its return values but all language representations only return void.
+        if let returns = returns, traitsWithNonVoidReturnValues.isEmpty {
+            diagnosticEngine.emit(extraReturnsProblem(returns, symbolKind: symbolKind))
+        }
+        return variants
+    }
+    
+    // MARK: Helpers
+    
+    /// Checks if the symbol's documentation is inherited from another source location.
+    private func hasInheritedDocumentationComment(symbol: UnifiedSymbolGraph.Symbol) -> Bool {
+        let symbolLocationURI = symbol.documentedSymbol?.mixins.getValueIfPresent(for: SymbolGraph.Symbol.Location.self)?.uri
+        for case .sourceCode(let location, _) in docChunkSources {
+            // Check if the symbol has documentation from another source location
+            return location?.uri != symbolLocationURI
+        }
+        // If the symbol didn't have any in-source documentation, check if there's a extension file override.
+        return docChunkSources.isEmpty
+    }
+    
+    private typealias Signatures = [DocumentationDataVariantsTrait: SymbolGraph.Symbol.FunctionSignature]
+    
+    /// Returns the symbol's function signatures for each variant trait, or `nil` if the symbol doesn't have any function signature data.
+    private static func traitSpecificSignatures(_ symbol: UnifiedSymbolGraph.Symbol) -> Signatures? {
+        var signatures: [DocumentationDataVariantsTrait: SymbolGraph.Symbol.FunctionSignature] = [:]
+        for (selector, mixin) in symbol.mixins {
+            guard let signature = mixin.getValueIfPresent(for: SymbolGraph.Symbol.FunctionSignature.self) else {
+                continue
+            }
+            signatures[DocumentationDataVariantsTrait(for: selector)] = signature
+        }
+        return signatures.isEmpty ? nil : signatures
+    }
+    
+    /// Checks if the language specific function signatures describe a throwing function in Swift that bridges to an Objective-C method with a trailing error parameter.
+    private static func hasSwiftThrowsObjectiveCErrorBridging(_ signatures: Signatures) -> Bool {
+        guard let objcSignature = signatures[.objectiveC],
+              objcSignature.parameters.last?.name == "error",
+              objcSignature.returns != knownVoidReturnValues[.objectiveC]!
+        else {
+            return false
+        }
+        guard let swiftSignature = signatures[.swift],
+                swiftSignature.parameters.last?.name != "error"
+        else {
+            return false
+        }
+        
+        return true
+    }
+    
+    /// Checks if the validator should synthesize documentation for an Objective-C error parameter.
+    private static func shouldAddObjectiveCErrorParameter(_ signatures: Signatures, _ parameters: [Parameter]?) -> Bool {
+        parameters?.last?.name != "error" && hasSwiftThrowsObjectiveCErrorBridging(signatures)
```

Comment:
Is it possible for some other Obj-C parameter to be called "error"? If that is possible should we check for that before appending the new, synthesized error parameter? What would happen if we end up with two duplicate error parameters?